data-model/src/main/java/io/micronaut/data/model/AssociationUtils.java

```diff
@@ -26,9 +33,50 @@
  * @since 1.0.0
  */
 @Internal
-class AssociationUtils {
+public final class AssociationUtils {
     /**
      * The split pattern for camel case.
      */
     static final Pattern CAMEL_CASE_SPLIT_PATTERN = Pattern.compile("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])");
+
+    private AssociationUtils() { }
+
+    /**
+     * Gets the  join paths from the method.
+     * @param method the method
+     * @return the join paths, if none defined and not of type FETCH then an empty set
+     */
+    public static Set<JoinPath> getJoinFetchPaths(ExecutableMethod<?, ?> method) {
```

Comment:
Please use `AnnotationMetadata` instead of `ExecutableMethod`

Reply:
Done